### PR TITLE
Adjust ansi grant statement to also handle snowflake grants.

### PR DESCRIPTION
**Pull request by mhaley-miovision**
This will close https://github.com/sqlfluff/sqlfluff/issues/538.

I would also like to update the revoke part too, but I will hold off on that until we get a first pass on the reworked grant statement. I chose to update the ansi dialect as suggested in the issue instead of making a new subclass.

### CODE REVIEW

File: src/sqlfluff/core/dialects/dialect_ansi.py [hunk 1]
```diff
@@ -1617,44 +1617,130 @@ class AccessStatementSegment(BaseSegment):
...
+    ]
+
+    _schema_object_types = OneOf(
+        *_schema_object_names,
+        Sequence("MATERIALIZED", "VIEW"),
+        Sequence("EXTERNAL", "TABLE"),
+        Sequence("FILE", "FORMAT"),
+    )
+
+    _schema_object_types_pural = OneOf(
```
Comment (NiallRees) [inline src/sqlfluff/core/dialects/dialect_ansi.py:1656]:
```suggestion
    _schema_object_types_plural = OneOf(
```

**Comment by NiallRees** [comment]
This looks really good - could you add a couple more comments explaining where the logic has come from, and that you decided to add the Snowflake dialect into the ANSI dialect, and why? I think that would help those coming to the logic in the future! Thanks again @mhaley-miovision

**Review by alanmcruickshank — APPROVED** [review]
Apart from the typo that @NiallRees found, I like this approach.

A little more documentation would be good, and we should follow up with the `REVOKE` clause pretty swiftly, because there's a lot of bits that we can reuse between the two. Happy for you to add to this PR before merging to do those things, or to make a seperate PR for them.

Nice approach @mhaley-miovision !

**Comment by mhaley-miovision** [comment]
Awesome, thanks for the reviews @NiallRees and @alanmcruickshank. Let's get this in as is, and over the next couple of days I can get revoke done. Sound like a plan?

**Comment by mhaley-miovision** [comment]
I have the revoke changes ready, they are based on these changes though, so once we get these in, I will get the other PR up right away.

**Review by NiallRees — APPROVED** [review]
Let's ⛴

**Comment by NiallRees** [comment]
Thank you again @mhaley-miovision, this is ace!

**Pull request merged**